azure/azure.go

```diff
@@ -114,6 +241,87 @@ func (a *azureOps) InstanceID() string {
 	return a.instance
 }
 
+func (a *azureOps) InspectInstance(instanceID string) (*cloudops.InstanceInfo, error) {
+
+	instInfo := &cloudops.InstanceInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			Name:   a.instance,
+			ID:     a.instance,
+			Zone:   "",
+			Region: "",
+		},
+	}
+	return instInfo, nil
+}
+
+func (a *azureOps) InspectInstanceGroupForInstance(instanceID string) (*cloudops.InstanceGroupInfo, error) {
+
+	ctx := context.Background()
+	agentPool, err := a.agentPoolsClient.Get(ctx, a.resourceGroupName, a.managedClusterName, a.agentPoolName)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get node pool details for [%s] of cluster [%v] in [%v] resource group."+
+			"Error: [%v] ", a.agentPoolName, a.managedClusterName, a.resourceGroupName, err)
+	}
+
+	// TODO: Currently multi-availability feature is in preview and
+	// zone field is not populated by azure agentPoolClient.
+	// For now, populate it with dummy value.
+	retval := &cloudops.InstanceGroupInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			ID:   *agentPool.ID,
+			Name: *agentPool.Name,
+		},
+		Zones: []string{"dummyzone"},
+	}
+
+	return retval, nil
+}
+
+func (a *azureOps) GetInstanceGroupSize(instanceGroupID string) (int64, error) {
+
+	ctx := context.Background()
+	agentPool, err := a.agentPoolsClient.Get(ctx, a.resourceGroupName, a.managedClusterName, instanceGroupID)
+	if err != nil {
+		return 0, err
+	}
+	return int64(*agentPool.Count), nil
+}
+
+// SetInstanceGroupSize sets desired node count per availability zone
+// for given instance group
+func (a *azureOps) SetInstanceGroupSize(instanceGroupID string,
+	count int64,
+	timeout time.Duration) error {
+
+	ctx := context.Background()
+	var cancel context.CancelFunc
+	if timeout > time.Nanosecond {
+		// If timeout is non-zero then
+		// wait infinitely for operation to complete
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
+
+	instanceGroupSize := int32(count)
+	agentPoolProperties := containerservice.ManagedClusterAgentPoolProfileProperties{
```

Comment:
Does the update work properly when you are not sending the other original values in the AgentPool object?
Should we not do a get and only update the count and call CreateOrUpdate?

Reply:
@piyush-nimbalkar yah, update is working properly here just with the properties to be updated( size in this case.)